src/driver/clint.rs

```diff
@@ -18,32 +14,32 @@ pub const MSIP_WIDTH: Width = Width::Byte4;
 pub const MTIMECMP_WIDTH: Width = Width::Byte8;
 pub const _MTIME_WIDTH: Width = Width::Byte8;
 
-#[derive(Clone, Debug)]
-pub struct ClintDriver {
-    /// The base address of the physical CLINT.
-    base: usize,
-}
```

Comment:
We want to keep this structure.

Also a note on performance, it might seem faster to not have a `base` field se we can use a constant. Turns out it is not necessarily the case. On RISC architecture loading a constant is expensive in terms of instructions, because instructions are 4 bytes and can only load a few bits at a time, so loading a constant ends-up taking ~4 instructions, whereas loading the value from a struct takes 1. Of course, there are caches, so the only way to know which is faster is to benchmark. I am just raising the point that's is not necessarily a bad thing to keep the base in the struct performance-wise, and it gives us more flexibility.